xrootd/client/client.go

```diff
@@ -42,10 +42,10 @@ type Client struct {
 	signRequirements protocol.SignRequirements
 }
 
-// NewClient creates a new xrootd client that connects to the given address.
+// NewClient creates a new xrootd client that connects to the given address using username.
 // When the context expires, a response handling is stopped, however, it is
 // necessary to call Cancel to correctly free resources.
-func NewClient(ctx context.Context, address string) (*Client, error) {
+func NewClient(ctx context.Context, address string, username string) (*Client, error) {
```

Comment:
I am wondering whether we shouldn't/couldn't use "functional options" here (so the number of parameters can be limited in simple cases, instead of passing `NewClient(ctx, "example.com:9999", "")` when identification, authentication and authorization aren't required)
although, in the real world, identification+authentication *is* required...

let's leave it like that for the moment.